src/equations/bbm_bbm_1d.jl

```diff
@@ -338,8 +338,7 @@ function create_cache(mesh, equations::BBMBBMEquations1D{BathymetryFlat},
 
     # homogeneous Neumann boundary conditions
     if solver.D1 isa DerivativeOperator ||
-       solver.D1 isa UniformCoupledOperator ||
-       solver.D1 isa FourierDerivativeOperator
+       solver.D1 isa UniformCoupledOperator
```

Comment:
This (and the other changes in this file) are unrelated, but I think we shouldn't allow a periodic `FourierDerivativeOperator` for solving a problem with reflecting boundary conditions.